### PR TITLE
Changing contract address from operator to registry

### PR DESCRIPTION
**Pull request by Luiserebii**
Hello,

I spent time looking through the plasma-core project to ensure that I could insert this in a clean way which would wrap around most of the other components. I wasn't sure whether to make a new service and extend base-service/base-provider, but concluded it may be overkill for the moment.

There is also room for a registry contract address fallback, which can be defined. I have placed a random address there for now, I thought it might be good to hardcode for safety? This would be for a safe default contract address that can be queried. 

Please let me know if there's anything more I can do, I tried to make sure I maximized this addition.

Tasks completed:
[✓] Query contract address from registry.
[✓] Remove old method of querying from operator.

Thank you!
Luiserebii

### CODE REVIEW

**Comment by kfichter** [comment]
Hi, thanks for making this PR so quickly! I’m taking today off (decompressing after a few days of insane work launching last week) but will review and hopefully merge tomorrow :-)

File: src/services/contract/http-provider.js [hunk 1]
```diff
@@ -1,8 +1,16 @@
 const BigNum = require('bn.js')
 const BaseContractProvider = require('./base-provider')
 const compiledContracts = require('plasma-contracts')
+//Default registryAddr for fallback. Value here is arbitrarily picked.
+const defaultregistryAddr = '0xA1f90e4933F9AF055e4a309DB54316552E7Fd10c'
+//registryAddr contains fallback to default
+const registryAddr = this.options.registryAddress || defaultregistryAddr
```
[the remark this answers is not in the document]
Comment (Luiserebii) [reply src/services/contract/http-provider.js]: That sounds reasonable, and definitely more OOP; I will try to work it into something like you provided.

File: src/services/contract/http-provider.js [hunk 2]
```diff
@@ -50,14 +58,11 @@ class HttpContractProvider extends BaseContractProvider {
   }
 
   async initContractAddress () {
-    // TODO: Replace this because we should really be getting
-    // the address from the registry.
-    if (this.services.operator.getEthInfo) {
-      await this.services.operator.waitForConnection()
-      const ethInfo = await this.services.operator.getEthInfo()
-      this.contract.options.address = ethInfo.plasmaChainAddress
+    if(plasmaChainName){
```
[the remark this answers is not in the document]
Comment (Luiserebii) [reply src/services/contract/http-provider.js]: Ahhh, ok; looking over the code, I was initially confused where the options were defined, but looking into it more carefully, I think I understand a lot better. Apologies, I was in the Java mindset of instance variables being declared initially at the head of the class, I'm not sure how I missed this, haha.

File: src/services/contract/http-provider.js [hunk 3]
```diff
@@ -50,14 +58,11 @@ class HttpContractProvider extends BaseContractProvider {
...
+      const contractAddress = await registryContract.methods.plasmaChainNames(plasmaChainName).call()
```
[the remark this answers is not in the document]
Comment (Luiserebii) [reply src/services/contract/http-provider.js]: Yup!

File: src/services/contract/http-provider.js [hunk 4]
```diff
@@ -1,8 +1,16 @@
...
 const compiledContracts = require('plasma-contracts')
+//Default registryAddr for fallback. Value here is arbitrarily picked.
+const defaultregistryAddr = '0xA1f90e4933F9AF055e4a309DB54316552E7Fd10c'
+//registryAddr contains fallback to default
+const registryAddr = this.options.registryAddress || defaultregistryAddr
+const plasmaChainName = this.options.plasmaChainName
 const plasmaChainCompiled = compiledContracts.plasmaChainCompiled
+const registryCompiled = compiledContracts.plasmaRegistryCompiled
 const erc20Compiled = compiledContracts.erc20Compiled
+const registryContract = new web3.eth.Contract(registryCompiled.abi, registryAddress)
```
[the remark this answers is not in the document]
Comment (Luiserebii) [reply src/services/contract/http-provider.js]: Ahhh, ok, looking more into it, I see where web3 is defined.

**Comment by Luiserebii** [comment]
Hello, thank you for the suggestions! I will examine the project's structure more carefully to make sure the changes work later today. Hopefully, they will pass!

File: src/services/contract/http-provider.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (kfichter) [reply src/services/contract/http-provider.js]: No worries at all! It’s definitely a little confusing, I should’ve explained how the services work :-) that’s a good idea for something to throw in the docs.

**Comment by kfichter** [comment]
Sounds good! Thanks for all the hard work 🙂

**Comment by Luiserebii** [comment]
Hello, before I make my changes, I just wanted to double-check on the design decisions so I can merge it cleanly in.

I notice that `const erc20Compiled = compiledContracts.erc20Compiled` is a constant defined outside of the class, which it is dependent on; is there a conceptual difference between these and psuedo-properties defined as get functions? (e.g. get address, get web3)

Also, are the get functions for variables to be reused so they can be referred to easily? Or are they supposed to just be constant properties disguised as a function?

Finally, how does the outer const and get function work in relation to the normal instance variables defined in the constructor? 

So, to summarize, I'm mainly just asking how `this.x = 1`, `const x = 1`, and `get x () { return 1; }` sort of conceptually work in the code. Apologies if there is something JavaScript-specific I'm missing. Just want to make sure I write something that fits in. Thank you!

**Comment by Luiserebii** [comment]
Hello, I made the needed changes!

I realized that defaultOptions was set to be overridden if an option is already set in the BaseService constructor, and moved it over to plasma.js, keeping the "import" flow. I ultimately decided that to keep it cleaner, to take registryAddress and plasmaChainName and place them as get properties. Part of my reasoning was to keep the general flow with the other get properties returning useful options to be used by the class.

The only minor thing that could maybe change is the hardcoding of the defaultRegistryAddress within plasma.js, I noticed that many of them were imported over from other files in the same directory. I might have to look deeper into the project to make that sort of design decision, so I decided to just leave it there.

Please let me know if there's anything else that could use a change, and thank you so much for your thorough reply! It definitely helped me decide on how to proceed with things.
